src/mintpy/prep_nisar.py

```diff
@@ -0,0 +1,411 @@
+
+    print(f'update mode: {inps.update_mode}')
+
+    input_files = sorted(glob.glob(inps.input_glob))
+    print(f"Found {len(input_files)} unwrapped files")
+
+    if inps.subset_lat:
+        bbox = (inps.subset_lat[0], inps.subset_lon[0], inps.subset_lat[1], inps.subset_lon[1])
+    else:
+        bbox=None
+
+    # extract metadata
+    metadata, bounds = extract_metadata(input_files, bbox)
+
+    # output filename
+    stack_file = os.path.join(inps.out_dir, "inputs/ifgramStack.h5")
+    geometry_file = os.path.join(inps.out_dir, "inputs/geometryGeo.h5")
+
+    # get date info: date12_list
+    date12_list = _get_date_pairs(input_files)
+
+    metadata = prepare_geometry(
+        outfile=geometry_file,
+        metaFile=input_files[0],
+        bbox=bounds,
+        metadata=metadata,
+        demFile=inps.dem_file
+        )
+
+    prepare_stack(
+        outfile=stack_file,
+        inp_files=input_files,
+        metadata=metadata,
+        bbox=bounds,
+        date12_list=date12_list,
+    )
+
+    print("Done.")
+
+    return
+
+
+def extract_metadata(input_files, bbox=None, polarization='HH'):
+    """Extract NISAR metadata for MintPy."""
+    meta_file = input_files[0]
+    meta = {}
+
+    # update dataset
+    for key, value in DATASETS.items():
+        if value:
+            value = value.replace("POL", polarization)
+            DATASETS[key] = value
+
+    with h5py.File(meta_file, 'r') as ds:
+        pixel_height = ds[DATASETS['ySpacing']][()]
+        pixel_width = ds[DATASETS['xSpacing']][()]
+        x_origin = min(ds[DATASETS['xcoord']][()])
+        y_origin = min(ds[DATASETS['ycoord']][()])
+        xcoord = ds[DATASETS['xcoord']][()]
+        ycoord = ds[DATASETS['ycoord']][()]
+        meta["EPSG"] = ds[DATASETS['epsg']][()]
+        meta['WAVELENGTH'] = LIGHTSPEED / ds[PROCESSINFO['centerFrequency']][()]
+        meta["ORBIT_DIRECTION"] = ds[PROCESSINFO['orbit_direction']][()].decode('utf-8')
+        meta['POLARIZATION'] = polarization
+        meta["ALOOKS"] = ds[DATASETS['azimuth_look']][()]
+        meta["RLOOKS"] = ds[DATASETS['range_look']][()]
+        meta['PLATFORM'] = ds[PROCESSINFO['platform']][()].decode('utf-8')
+        meta['STARTING_RANGE'] = min(ds[PROCESSINFO['rdr_slant_range']][()].flatten())
+        start_time = datetime.datetime.strptime(ds[PROCESSINFO['start_time']][()].decode('utf-8')[0:26],
+                                                "%Y-%m-%dT%H:%M:%S.%f")
+        end_time = datetime.datetime.strptime(ds[PROCESSINFO['end_time']][()].decode('utf-8')[0:26],
+                                              "%Y-%m-%dT%H:%M:%S.%f")
+
+    t_mid = start_time + (end_time - start_time) / 2.0
+    meta["CENTER_LINE_UTC"] = (
+            t_mid - datetime.datetime(t_mid.year, t_mid.month, t_mid.day)
+    ).total_seconds()
+    meta["X_FIRST"] = x_origin
+    meta["Y_FIRST"] = y_origin
+    meta["X_STEP"] = pixel_width
+    meta["Y_STEP"] = pixel_height
```

Comment:
Two thoughts on this-
1. perhaps we could get these from like `dataset.GetGeotransform()`, if these are all geocoded. but if it's a mix of radar/geo, i can see why you'd do it this way
2. I think we need to add 1/2 a pixel to make these compliant with the top-left-edge convention, since I believe the `xCoordinates` refer to pixel centers

Reply:
Thank you @scottstanie for reviewing and all the good suggestions. 
For the first one I think it is a mixed data and might not be possible.
Second one, you are right I will check on that again